### PR TITLE
using ui-router resolve to load user profile data

### PR DESCRIPTION
**Pull request by 2is10**
re-applying #13296 with a dependency injection fix
@yipingliao

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/route.js [hunk 1]
```diff
@@ -69,6 +69,12 @@ angular.module('kifi')
         url: '/:username',
         templateUrl: 'userProfile/userProfile.tpl.html',
         controller: 'UserProfileCtrl',
+        resolve: {
+          userProfileActionService: 'userProfileActionService',
+          profile: ['userProfileActionService', '$stateParams', function (userProfileActionService, $stateParams) {
+            return userProfileActionService.getProfile($stateParams.username);
+          }]
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/route.js:76]: This fix is using quoted strings for the names of the injected arguments here, as we do elsewhere.

**Comment by 2is10** [comment]
The protractor test that failed before passes with this fix, and I’ve manually verified that user profile navigations work by pausing the protractor tests using `browser.pause()`.

File: kifi-backend/modules/shoebox/angular/src/route.js (hunk 1 above)
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/route.js:76]: Ah, that's why it breaks with prod assets!

**Comment by yipingliao** [comment]
:+1:

File: kifi-backend/modules/shoebox/angular/src/route.js (hunk 1 above)
Comment (2is10) [inline kifi-backend/modules/shoebox/angular/src/route.js:76]: @yipingliao Yeah, unfortunately, none of the ui-router examples I saw used this approach, so I thought it must somehow be exempt. Silly me.

**Pull request merged**